gpio_test.go

```diff
@@ -0,0 +1,158 @@
+package gpio
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// valudate and extract test setup
+func getGpioSetup(t *testing.T) (gpiodev string, gpioin uint32, gpioout uint32) {
+	if len(os.Getenv("GPIO_DEV_PATH")) == 0 {
+		t.Skip("Please set GPIO_DEV_PATH, GPIO_TEST_PIN to run tests ")
+		return "", 0, 0
+	}
+	if len(os.Getenv("GPIO_TEST_PIN")) == 0 {
+		t.Skip("Please set GPIO_DEV_PATH, GPIO_TEST_PIN to run tests ")
+		return "",0, 0
+	}
+	testPinIn, err := strconv.Atoi(os.Getenv("GPIO_TEST_PIN"))
+	if (testPinIn < 1 || err != nil) {
```

Comment:
Pretty sure it's not the scope of current library to validate pin numbers people actually may have.

Reply:
For some reason I thought that GPIO pins start at 1, I'll go and fix that assumption